app/routes.py

```diff
@@ -1 +1,239 @@
-from flask import Blueprint
\ No newline at end of file
+import os
+from datetime import datetime
+import requests
+from app import db
+
+from flask import Blueprint, jsonify, make_response, request, abort
+
+from app.models.task import Task
+from app.models.goal import Goal
+
+
+# DEFINING THE BLUEPRINTS
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+# error handling
+def validate_model(cls, model_id):
+    try:
+        model_id = int(model_id)
+    except:
+        abort(make_response({"message":f"task {model_id} invalid"}, 400))
+
+    model = cls.query.get(model_id)
+
+    if not model:
+        abort(make_response({"details": f"{cls.__name__} id {model_id} not found"}, 404))
+
+    return model
+
+
+# ============================= CRUD ROUTES  =============================
+# CREATE TASK- POST request to /tasks
+@tasks_bp.route("", methods=["POST"])
+def create_task():
+    request_body = request.get_json()
+    
+    if "title" not in request_body or "description" not in request_body:
```

Comment:
Like we validated models, could we also make a general function that validates request bodies? Here's an example:
```python
def validate_request_body(request_body, keys):
    for key in keys:
        if not request_body.get(key): 
            abort(make_response({
                'Invalid Data': f'missing key: {key}'
            }, 400))

    return True
```
We can pass in the `request_body` and a list of strings that are `keys` and then check to see if those keys are present.